cli/pack/common.go

```diff
@@ -0,0 +1,553 @@
+	"^\\.$",       // excludes a hard link to the current directory.
+	"\\w+.yml",    // excludes all yml files
+	"\\w+.yaml",   // excludes all yaml files
+	"\\w+.tar.gz", // excludes all tarballs
+}
+
+// pack prepares a temporary directory for packing.
+// Returns a path to the prepared directory or error if it failed.
+func pack(cmdCtx *cmdcontext.CmdCtx) (string, error) {
+	var err error
+	packCtx := &cmdCtx.Pack
+
+	startPath, _ := os.Getwd()
+	defer func() {
+		_ = os.Chdir(startPath)
+	}()
+	err = goToEnvRoot(packCtx)
+	if err != nil {
+		return "", err
+	}
+
+	// Create temporary directory step.
+	basePath, err := os.MkdirTemp("", "tt_pack")
+	if err != nil {
+		return "", err
+	}
+
+	defer func() {
+		if err != nil {
+			err := os.RemoveAll(basePath)
+			log.Warnf("Failed to remove a directory %s: %s", basePath, err.Error())
+		}
+	}()
+
+	err = createPackageStructure(basePath)
+	if err != nil {
+		return "", err
+	}
+
+	// Copy binaries step.
+	if cmdCtx.Pack.App.BinDir != "" &&
+		((!cmdCtx.Pack.TarantoolIsSystem && !cmdCtx.Pack.WithoutBinaries) ||
+			cmdCtx.Pack.WithBinaries) && !cmdCtx.Pack.RpmDeb.WithTarantoolDeps {
+		baseEnvBinPath := filepath.Join(basePath, envPath, binPath)
+		err = copy.Copy(packCtx.App.BinDir, baseEnvBinPath)
+		if err != nil {
+			return "", err
+		}
+	}
+
+	// Copy modules step.
+	if packCtx.ModulesDirectory != "" {
+		baseEnvModulesPath := filepath.Join(basePath, envPath, modulesPath)
+		err = copy.Copy(packCtx.ModulesDirectory, baseEnvModulesPath)
+		if err != nil {
+			return "", err
+		}
+	}
+
+	// Collect app list step.
+	appList := packCtx.AppList
+	if packCtx.AppList == nil && packCtx.App.InstancesEnabled != "." {
+		// Initialize a default list of strings for preparing a black list.
+		ExcludeListExpressions := prepareDefaultExcludeListExpressions()
+
+		excludeList, err := prepareExcludeList(ExcludeListExpressions)
+		if err != nil {
+			return "", fmt.Errorf("failed to compile regular expressions: %v", err)
+		}
+
+		appList, err = collectAppList(packCtx.App.InstancesEnabled, excludeList)
+		if err != nil {
+			return "", err
+		}
+	} else if packCtx.App.InstancesEnabled == "." {
+		// If running tt pack with instances_enabled: .
+		// it supposed to pack a current directory as an application.
+		appPath, err := os.Getwd()
+		if err != nil {
+			return "", err
+		}
+		appList = []string{filepath.Base(appPath)}
+		err = os.Chdir("..")
+		if err != nil {
+			return "", err
+		}
+	}
+
+	log.Infof("Apps to pack: %v", strings.Join(appList, " "))
+
+	// Copy all apps to a temp directory step
+	for _, appName := range appList {
+		err = copyAppSrc(packCtx, appName, basePath)
+		if err != nil {
+			return "", err
+		}
+
+		if packCtx.Archive.All {
+			err = copyArtifacts(packCtx, appName, basePath)
+			if err != nil {
+				return "", err
+			}
+		}
+
+		err = createAppSymlink(packCtx, appName, basePath)
+		if err != nil {
+			return "", err
+		}
+
+		err = buildRocks(cmdCtx, appName, basePath)
```

Comment:
Lets move it above create symlink step. There is need to have the app, which is not built yet, enabled.